pkg/provider/aws/resources/s3/bucket.go

```diff
@@ -17,12 +17,13 @@ type (
 		Name          string
 		ConstructsRef []core.AnnotationKey
 		ForceDestroy  bool
+		IndexDocument string
 	}
 )
 
-func NewS3Bucket(fs *core.Fs, appName string) *S3Bucket {
+func NewS3Bucket(fs core.Construct, appName string) *S3Bucket {
```

Comment:
yeah im doing it so we can pass either of those types. I thought about doing 2 functions, but nothing within the bucket is derived from constructs outside of the reference to it so i opted otherwise and deferred to having the provider just create the right resources.

Brings up an interesting case of mapping what constructs can create and if we want to restrict that